pkg/services/accesscontrol/resourcepermissions/service.go

```diff
@@ -66,7 +66,9 @@ func New(cfg *setting.Cfg,
 		for _, a := range actions {
 			actionSet[a] = struct{}{}
 		}
-		actionSetService.StoreActionSet(options.Resource, permission, actions)
+		if features.IsEnabled(context.Background(), featuremgmt.FlagAccessActionSets) {
+			actionSetService.StoreActionSet(options.Resource, permission, actions)
+		}
```

Comment:
Note: putting this behind a feature toggle - there shouldn't be anything risky with registering an action set, but better safe than sorry.